Source/WebCore/Modules/url-pattern/URLPattern.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "URLPattern.h"
+
+#include "URLPatternInit.h"
+#include "URLPatternOptions.h"
+#include "URLPatternResult.h"
+#include <wtf/RefCounted.h>
+
+namespace WebCore {
+
+WTF_MAKE_TZONE_OR_ISO_ALLOCATED_IMPL(URLPattern);
+
+Ref<URLPattern> URLPattern::create(const URLPatternInput, const String&& baseURL, URLPatternOptions&&)
+{
+    UNUSED_PARAM(baseURL);
+
+    return adoptRef(*new URLPattern());
+}
+
+Ref<URLPattern> URLPattern::create(std::optional<URLPatternInput>, URLPatternOptions&&)
+{
+    return adoptRef(*new URLPattern());
+}
+
+URLPattern::URLPattern() = default;
+
+URLPattern::~URLPattern() = default;
+
+ExceptionOr<bool> URLPattern::test(std::optional<URLPatternInput>, std::optional<String>&& baseURL) const
+{
+    UNUSED_PARAM(baseURL);
+
+    return Exception { ExceptionCode::NotSupportedError };
+}
+
+std::optional<URLPatternResult> URLPattern::exec(std::optional<URLPatternInput>&&, std::optional<String>&& baseURL) const
```

Comment:
This should return `ExceptionOr<std::optional<URLPatternResult>>` (like the function above)